app/src/main/java/org/astraea/app/publisher/MetricPublisher.java

```diff
@@ -63,184 +44,33 @@ public static void main(String[] args) {
-        .property(NAME_PROPERTY, "BeanFetch")
-        .attribute(RATE_PROPERTY, Double.class, () -> fetchRateSensor.measure(RATE_PROPERTY))
-        .description("MBeans fetch-rate since publisher start up. (beans/second)")
-        .register();
-
-    var JMXFetcherThreads =
-        jmxFetcherThreads(
-            3, targetClients, arguments.period, beanQueue, fetchRateSensor, close::get);
-    var publisherThreads =
-        publisherThreads(
-            2, arguments.bootstrapServers(), beanQueue, () -> close.get() && beanQueue.isEmpty());
-    var periodicJobPool = Executors.newScheduledThreadPool(1);
-    var threadPool = Executors.newFixedThreadPool(3 + 2);
     var admin = Admin.of(arguments.bootstrapServers());
-
-    // Periodically update MBeanClient. (MBeanClients may change when broker added into cluster)
-    var periodicUpdate =
-        periodicJobPool.scheduleAtFixedRate(
-            () ->
-                admin
-                    .nodeInfos()
-                    .thenAccept(
-                        nodes -> {
-                          // TODO: Real elements in queue may not consistent with `inQueue` object.
-                          // Other threads (JMXFetcherThreads) may add element to this queue (as
-                          // they complete the fetch). This may result in duplicate targets in the
-                          // DelayQueue (`targetClients`). Discussion:
-                          // https://github.com/skiptests/astraea/pull/1481#discussion_r1112972154
-                          var inQueue =
-                              targetClients.stream()
-                                  .map(target -> target.id)
-                                  .collect(Collectors.toUnmodifiableSet());
-                          nodes.stream()
-                              .filter(node -> !inQueue.contains(String.valueOf(node.id())))
-                              .forEach(
-                                  node ->
-                                      targetClients.put(
-                                          new DelayedIdClient(
-                                              arguments.period,
-                                              String.valueOf(node.id()),
-                                              MBeanClient.jndi(
-                                                  node.host(),
-                                                  arguments.idToJmxPort().apply(node.id())))));
-                        }),
-            0,
-            5,
-            TimeUnit.MINUTES);
-
-    // All JMXFetchers keep fetching on targets
-    JMXFetcherThreads.forEach(threadPool::execute);
-    // All publishers keep publish mbeans
-    publisherThreads.forEach(threadPool::execute);
-
-    // Run until the given time-to-live passed
-    try {
-      Thread.sleep(arguments.ttl.toMillis());
-    } catch (InterruptedException ie) {
-      ie.printStackTrace();
+    var topicSender = new TopicSender(arguments.bootstrapServers());
```

Comment:
Could you add helper to `Sender`? `Sender.toTopic(String)` is the better entry point.